### PR TITLE
[13.0][MIG] base_user_locale

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Supersedes https://github.com/OCA/server-ux/pull/158.

Tests are a bit refactored due to https://github.com/odoo/odoo/pull/38357.

### CODE REVIEW

**Review by LoisRForgeFlow — APPROVED** [review]
functional review in local, LGTM :+1:

File: base_user_locale/controllers/web_client.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+from odoo.http import request, route
+
+from odoo.addons.web.controllers.main import WebClient, ensure_db
+
+
+class WebClient(WebClient):
+    def get_user_lang_parameters(self, user):
+        res = {}
+        date_format = user.date_format or user.company_id.date_format
```
Comment (LoisRForgeFlow) [inline base_user_locale/controllers/web_client.py]: all these `user.company_id` should be `user.env.company`, isn't it?

Comment (MiquelRForgeFlow) [reply base_user_locale/controllers/web_client.py]: right

**Review by AaronHForgeFlow — APPROVED** [review]
Tested on runbot and working fine. Nevertheless, it would be great a configuration section in the readme. Currently I have to go to the code in order to know how to configure things.

On the roadmap perhaps we can include a help section so the users can configure their own settings on their own.

**Comment by LoisRForgeFlow** [comment]
Readme updated! thanks for your review :)

/ocabot merge nobump

**Pull request merged**